components/base/fake/fake.go

```diff
@@ -0,0 +1,160 @@
+// Package fake implements a fake base.
+package fake
+
+import (
+	"context"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/base"
+	limo "go.viam.com/rdk/components/base/agilex"
+	"go.viam.com/rdk/components/base/boat"
+	"go.viam.com/rdk/components/base/wheeled"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// Config is used for converting config attributes.
+type Config struct {
+	BaseModel string `json:"base-model,omitempty"`
+}
+
+func init() {
+	registry.RegisterComponent(
+		base.Subtype,
+		resource.NewDefaultModel("fake"),
+		registry.Component{
+			Constructor: func(
+				ctx context.Context,
+				_ registry.Dependencies,
+				config config.Component,
+				logger golog.Logger,
+			) (interface{}, error) {
+				return &Base{Name: config.Name}, nil
```

Comment:
[minor] Since this is actually doing something now, let's use `NewBase` as a constructor since we don't have two points of  truth for the registry and internal to RDK.
```suggestion
				return NewBase(ctx, config, logger)
```